File: mobile/lib/features/routes/domain/route_model.dart
import '../../dispatch/domain/vehicle.dart';
import 'route_stop.dart';

class RouteModel {
  const RouteModel({
    required this.id,
    required this.code,
    required this.status,
    required this.stops,
    this.vehicle,
    this.driver,
  });

  factory RouteModel.fromJson(Map<String, dynamic> json) => RouteModel(
        id: json['id'] as int,
        code: json['code'] as String,
        status: json['status'] as String,
        vehicle: json['vehicle'] != null ? Vehicle.fromJson(json['vehicle'] as Map<String, dynamic>) : null,
        driver: json['driver'] != null ? UserOption.fromJson(json['driver'] as Map<String, dynamic>) : null,
        stops: (json['stops'] as List? ?? [])
            .map((e) => RouteStop.fromJson(e as Map<String, dynamic>))
            .toList(),
      );

  final int id;
  final String code;
  final String status;
  final Vehicle? vehicle;
  final UserOption? driver;
  final List<RouteStop> stops;
}

File: backend/app/Models/RouteModel.php
<?php

namespace App\Models;

use App\Enums\RouteStatus;
use App\Support\Auditable;
use App\Support\BelongsToCompany;
use Illuminate\Database\Eloquent\Model;

/**
 * Se nombra RouteModel (no Route) para no chocar con
 * Illuminate\Routing\Route, muy usado en todo el framework.
 */
class RouteModel extends Model
{
    use Auditable, BelongsToCompany;

    protected $table = 'routes';

    protected $fillable = [
        'company_id', 'code', 'vehicle_id', 'driver_id', 'status', 'started_at', 'finished_at',
    ];

    protected $casts = [
        'status' => RouteStatus::class,
        'started_at' => 'datetime',
        'finished_at' => 'datetime',
    ];

    public function vehicle()
    {
        return $this->belongsTo(Vehicle::class);
    }

    public function driver()
    {
        return $this->belongsTo(User::class, 'driver_id');
    }

    public function stops()
    {
        return $this->hasMany(RouteStop::class, 'route_id')->orderBy('sequence');
    }
}
